yaml_edit/lib/src/strings.dart

```diff
@@ -279,7 +279,7 @@ String yamlEncodeBlockString(
       final formattedValue =
           yamlEncodeBlockString(entry.value, newIndentation, lineEnding);
 
-      if (isCollection(entry.value)) {
+      if (isCollection(entry.value) && !isEmpty(entry.value)) {
```

Comment:
Can we add a comment that:
`// Empty collections are always encoded in flow-style, so new-line must be avoided`

I assume this is why we have this hack?

Reply:
Yeap! and added!